pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1035,9 +1035,14 @@ func (r *ReconcileHyperConverged) migrateBeforeUpgrade(req *common.HcoRequest) (
 		return false, err
 	}
 
+	oldMetricsSVCRemoved, err := r.removeOldMetricsObjs(req)
+	if err != nil {
+		return false, err
+	}
+
 	removeOldQuickStartGuides(req, r.client, r.operandHandler.GetQuickStartNames())
 
-	return upgradePatched, nil
+	return upgradePatched || oldMetricsSVCRemoved, nil
```

Comment:
I believe this is not needed. This function returns `true` if the **`HyperConverged`** CR is modified, in order to requeue later. This is not neede when removing another resources, as services or endpoints. I think the `removeOldMetricsObjs` function should only return error, but not the boolean value.